Feature sign in

LaithJabali opened the pull request.

LaithJabali committed:
Fix some problems in the code

LaithJabali committed:
Finish login page

And add all its components, which will be used on the Sign Up page

LaithJabali committed:
Finish Sign Up Page

iamhaitham commented on src/components/Header/data.js:
```diff
@@ -6,12 +6,12 @@ export const navLinks = [
   },
   {
-    url:'/SignUp',
+    url:'/SignUpPage',
```
Why have you added the word "Page" to the url? 
You don't see that usually in any website's url.

iamhaitham commented on src/components/pages/routes.js:
```diff
@@ -2,8 +2,8 @@ const routes = {
-    signIn: '/SignIn',
-    signUp: '/SignUp',
+    signIn: '/SignInPage',
+    signUp: '/SignUpPage',
```
Usually you don't see websites with paths like these (You don't include the word "Page" in the path).
Please update both.

iamhaitham commented on src/components/Signs/HaveAccount.js:
```diff
@@ -0,0 +1,14 @@
+const HaveAccount = ({ DoU, Sign }) => {
+  return (
+    <div className={Style.signUpLink}>
+      {DoU}
```
What does DoU suppose to mean?
It is not a clear name for a prop.
Please give it a more meaningful name.

iamhaitham commented on src/components/Signs/HaveAccount.js:
```diff
@@ -0,0 +1,14 @@
+import Style from './Signs.module.css';
+import MyLink from './MyLink';
+
+const HaveAccount = ({ DoU, Sign }) => {
```
Props should be in camelCase not on PascalCase

iamhaitham commented on src/components/Signs/MyLink.js:
```diff
@@ -0,0 +1,16 @@
+  return (
+    <>
+      {Links.map((item) => (
+        <Link className={item.LinkClass} to={item.To}>
```
Please add a unique key to the returned component.

iamhaitham commented on src/components/Signs/SignIn.js:
This component is actually the same as SignUp.js
You are returning the same structure of code:
div -> div -> SignTitle -> SignForm -> HaveAccount
You can extract that structure to a separate component with the needed props, and in SignUp.js and SignIn.js pass the needed props.
This will eliminate the duplication between the two files.

iamhaitham commented on src/components/Signs/InputItemForm.js:
```diff
@@ -0,0 +1,17 @@
+    <>
+      {formItems.map((item) => (
+        <Form.Item key={item.name} name={item.name} rules={item.rules}>
+          <Input type={item.type} placeholder={item.placeholder} className={Style.input} />
```
type={item.type}
In most objects you have provided the type property, but you missed it in the email in both forms.

LaithJabali committed:
Updated the code based on the Monitor feedback

in Signs Components
new file with name : Sign.js To remove duplication in SignIn and SignUp
and Minor modifications to the rest of the files

iamhaitham commented on src/components/Signs/MyLink.js:
```diff
@@ -4,8 +4,8 @@ import { Link } from 'react-router-dom';
-      {Links.map((item) => (
-        <Link className={item.LinkClass} to={item.To}>
+      {Links.map((item, index) => (
+        <Link key={index} className={item.LinkClass} to={item.To}>
```
Please read the Pitfall section in [this link](https://react.dev/learn/rendering-lists#why-does-react-need-keys) to understand why this is not a good practice.
Please use another unique key.

LaithJabali committed:
Update MyLink Component

fix keys
